### ISSUE TITLE
Custom configuration gets overwritten when starting the container for the first time

### ISSUE THREAD

**Issue by prydonius** [bitnami/bitnami-docker-mariadb#65]
**Issue**

I created a directory on my host containing a custom `my.cnf` file, and mounted it using the following command:

```
docker run --rm -ti -v `pwd`/mariadb-conf:/bitnami/mariadb/conf bitnami/mariadb
```

**What I expected**
- MariaDB would pick up my custom configuration

**What happened**
- My custom `my.cnf` got overwritten with the image's default one

**Proposed solution**
- Don't overwrite `/bitnami/mariadb/conf/my.cnf`
- Keep the default config in `/opt/bitnami/mysql/my.cnf` and use the include directive to pick up custom settings:
```
!include /bitnami/mariadb/conf/my.cnf
```
(we may need to create it if it doesn't exist)

This will allow people to customise the MySQL configuration in the following ways:
- Building `FROM` this image and dumping a custom configuration in `/bitnami/mariadb/conf`
- Mounting `my.cnf` in `/bitnami/mariadb/conf` (this approach could cause conflict if also mounting the whole `/bitnami/mariadb` folder for persistence)

**Comment by devurandom** [comment]
Solution 2 (use of `!include`) seems preferable to me, so I don't need to figure out sensible defaults for everything, except were my application requires specific settings. Maybe a combination is possible, to allow more experienced users to override everything with their own settings? E.g. allow to specify a custom my.cnf config file location via envvar, but also add a conf.d directory that is included by the default my.cnf?

**Comment by prydonius** [comment]
Cool, yeah I agree that `include` makes the most sense. Every default should be overridable by custom config file (it will be included last), so it might not be necessary to have a combination. Can you see a specific use case where an include won't be sufficient?

**Comment by devurandom** [comment]
> Can you see a specific use case where an include won't be sufficient?

No, but I am quite a noob when it comes to MySQL / MariaDB settings. For me the `!include` variant is sufficient - I do want whatever defaults you ship, because I assume you know much better than me which settings are sensible.

**Comment by prydonius** [comment]
Closing as this was fixed in #66
